# Chatty

matthanfoo opened the pull request:
# Chatty

Chatty helps you keep track of the million things on your to-do list, but it's also 
- text-based
- easy to learn
- SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/matthanfoo/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

Here's what an example task list looks like:

> Here are the tasks in your list:
> 1. [T] [ ] borrow book
> 2. [D] [X] return book /by 05-06-2022 (by: 2022-06-05T00:00)
> 3. [E] [ ] project meeting /from 12-02-2025 14:00 /to 12-02-2025 17:00 (from: 2025-02-12T14:00 to: 2025-02-12T14:00)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Chatty {
   public static void main(String[] args) {
      new Chatty("ChattyData.csv").run();
   }
}
```
**TRY IT OUT NOW!**

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

matthanfoo committed:
Add Java files for initial stages

matthanfoo committed:
Add Week 2 Stage 4

matthanfoo committed:
Add delete function

Nano-233 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,152 @@
+        System.out.println("What can I do for you?");
+        ArrayList<Task> userInputs = new ArrayList<Task>();
+        try {
+            String s = br.readLine();
```
Should this have a more descriptive variable name? such as userInput

Nano-233 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,152 @@
+            String s = br.readLine();
+            while (s != null) {
+                System.out.println("____________________________________________________________");
+                if (s.equals("bye") || s.equals("Bye")) {
```
Would using `s.equalsIgnoreCase("bye")` here be more efficient and readable?

Nano-233 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,152 @@
+                    }
+                } else if (s.contains("mark")) {
+                    Pattern p = Pattern.compile("([0-9]+$)");
+                    Matcher m = p.matcher(s);
```
Any reason this two line definition is used multiple times instead of using a helper method?
(for example again in line 52-53)

Nano-233 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,152 @@
+                    } else {
+                        System.out.println("Invalid item");
+                        break;
+                    }
```
Should switch-case be used instead of this many if-else statements to increase readability and maintainability?

Nano-233 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,152 @@
+                                System.out.println("   " + task);
+                            }
+                        } else {
+                            System.out.println("Invalid item");
```
For error messages, should there be logics or other ways to implement them so that they are not  hardcoded?

Nano-233 commented on src/main/java/Chatty.java:
```diff
@@ -0,0 +1,152 @@
+
+                    } else {
+                        System.out.println("Invalid item");
+                        break;
```
This may break the app as it breaks out of the while loop

Nano-233 commented on src/main/java/Chatty.java:
Should this code be refactored to improve readability and logic so that there are less hardcoding involved?

Nano-233 reviewed (commented):
I like how many potential user errors are considered and accounted for. Perhaps the code could be cleaner by refactoring?

matthanfoo committed:
Update to Level 7

matthanfoo committed:
Add Level 8 functionality

matthanfoo committed:
Organize into more classes for more OOP

matthanfoo committed:
Add Week 3 Increments (OOP and Packages)

matthanfoo committed:
Merge branch 'add-gradle-support'

matthanfoo committed:
Merge branch 'branch-gradle'

matthanfoo committed:
Add JUnit testing capabilities

matthanfoo committed:
Add JavaDoc comments for all non-intuitive methods

matthanfoo committed:
Add find feature

matthanfoo committed:
Rewrite files in Storage.java slightly to match coding standard.

matthanfoo committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

matthanfoo committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/running/TaskList.java
#	src/main/java/tasks/Task.java

matthanfoo committed:
Update to match coding standard

matthanfoo committed:
Add JavaFX functionality

matthanfoo committed:
Delete obsolete code

matthanfoo committed:
Delete unused code

matthanfoo committed:
Add assertions

matthanfoo committed:
Remove long methods and over indentation, improve abstraction

matthanfoo committed:
Add recurring event functionality

matthanfoo committed:
Merge pull request #2 from matthanfoo/branch-A-Assertions

Branch-A-Assertions

matthanfoo committed:
Resolve merge conflicts

matthanfoo committed:
Resolve merge conflicts

Parser execute method currently has much more than 50 LoC
and has too many levels/layers of nesting

Add new methods to Parser class in order to increase level of abstraction and also reduce length of execute method

Add recur functionality to create repeated tasks at daily/weekly/monthly/yearly intervals for a specified number of repetitions

matthanfoo committed:
Add screenshot of basic UI functionality

Display list and mark functions

matthanfoo committed:
Update README.md

matthanfoo committed:
Update README.md

matthanfoo committed:
Merge branch 'master' into branch-A-CodeQuality

matthanfoo committed:
Merge pull request #3 from matthanfoo/branch-A-CodeQuality

Branch-A-CodeQuality
